Vagrantfile

```diff
@@ -0,0 +1,36 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+ENV["TERM"] = "xterm-256color"
+ENV["GOPATH"] = "/go"
+ENV["LC_ALL"] = "en_US.UTF-8"
+
+Vagrant.configure("2") do |config|
+  # config.vm.box = "fedora/25-cloud-base"
+  # config.vm.provision "shell", inline: "dnf install -y go git docker"
+  config.vm.box = "ubuntu/zesty64"
+  config.vm.provision "shell", inline: "DEBIAN_FRONTEND=noninteractive apt-get install -y golang git docker.io systemd-container tmux"
+
+  # config.vbguest.auto_update = true
+
+  config.vm.synced_folder ".", "/vagrant", disabled: true
+  config.vm.synced_folder ".", "/go/src/github.com/kinvolk/kubeadm-nspawn", create: true
+  # config.vm.provider :virtualbox do |vbox|
+  #     vbox.check_guest_additions = false
+  #     vbox.functional_vboxsf = false
+  # end
+
+  config.vm.provider :virtualbox do |vb|
+      vb.customize ["modifyvm", :id, "--memory", "1024"]
```

Comment:
Actually, the error was:

<details>

```
Env for linux/amd64: GOOS=linux GOARCH=amd64 GOROOT=/usr/local/go_k8s_patched CGO_ENABLED= CC=
# k8s.io/kubernetes/cmd/genman
/usr/local/go_k8s_patched/pkg/tool/linux_amd64/link: running gcc failed: fork/exec /usr/bin/gcc: cannot allocate memory

!!! [0614 16:33:34] Call tree:
!!! [0614 16:33:34]  1: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:715 kube::golang::build_binaries_for_platform(...)
!!! [0614 16:33:34]  2: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! Error in /go/src/k8s.io/kubernetes/hack/lib/golang.sh:530
  Error in /go/src/k8s.io/kubernetes/hack/lib/golang.sh:530. '((i<4-1))' exited with status 2
Call stack:
  1: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:530 kube::golang::build_binaries_for_platform(...)
  2: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:715 kube::golang::build_binaries(...)
  3: hack/make-rules/build.sh:27 main(...)
Exiting with status 1
!!! [0614 16:33:34] Call tree:
!!! [0614 16:33:34]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! Error in /go/src/k8s.io/kubernetes/hack/lib/golang.sh:711
  Error in /go/src/k8s.io/kubernetes/hack/lib/golang.sh:711. '((i<3-1))' exited with status 1
Call stack:
  1: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:711 kube::golang::build_binaries(...)
  2: hack/make-rules/build.sh:27 main(...)
Exiting with status 1
!!! [0614 16:33:34] Call tree:
!!! [0614 16:33:34]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! Error in /go/src/k8s.io/kubernetes/hack/lib/golang.sh:620
  Error in /go/src/k8s.io/kubernetes/hack/lib/golang.sh:620. '((i<3-1))' exited with status 1
Call stack:
  1: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:620 kube::golang::build_binaries(...)
  2: hack/make-rules/build.sh:27 main(...)
Exiting with status 1
Makefile:89: recipe for target 'all' failed
make: *** [all] Error 1
!!! [0614 16:33:35] Call tree:
!!! [0614 16:33:35]  1: build/../build/common.sh:513 kube::build::run_build_command_ex(...)
!!! [0614 16:33:35]  2: build/run.sh:30 kube::build::run_build_command(...)
!!! Error in build/../build/common.sh:578
  Error in build/../build/common.sh:578. '((i<4-1))' exited with status 2
Call stack:
  1: build/../build/common.sh:578 kube::build::run_build_command_ex(...)
  2: build/../build/common.sh:513 kube::build::run_build_command(...)
  3: build/run.sh:30 main(...)
Exiting with status 1
```

</details>

With the fork/exec error, I wonder if it is about pids limit in the pids cgroup controller. But it does not seem to be the case:

```
$ sudo cat  /sys/fs/cgroup/pids//docker/d23fd83a4c92fa1854f111cb77d39c70496589d0d8c3a115bba292b9bacffc57/pids.max
max
```

any ideas?